unittest_pythoncode.py

```diff
@@ -3,36 +3,33 @@
+        self.assertEqual(gityghasemi.python_bf(bf_code),'Mostafa Ahangarha')
 
-        bfcode='++++[->++++++++<]-[----->>++>++<<<]>>+++++++.>-.+++.<+++++.>-------.+++.<<.>>.+.+++.------.+++++++.-----.<<+.'
-        self.assertEqual(gityghasemi.execute(bfcode), 'mehrad dehbid!')
+        bf_code='++++[->++++++++<]-[----->>++>++<<<]>>+++++++.>-.+++.<+++++.>-------.+++.<<.>>.+.+++.------.+++++++.-----.<<+.'
+        self.assertEqual(gityghasemi.python_bf(bf_code), 'mehrad dehbid!')
 
-        bfcode='>+++++++[<++++++++++> -]<+.print G>> ++++++++[< +++++++++>-]<+.print I>>+++++++++++++++++++++[<++++>-]< .print T>> +++++++++++[< ++++++++> -]<+.print Y>++++++++++++++++++++++++++++++++.print ''>>+++++++[< ++++++++++> -]<+.print G>> ++++++++[< +++++++++> -]<. print H>> +++++++++++++[< +++++> -]<.print A>> +++++++++++++++++++++[< ++++> -]< -.print S>> +++++++++++++++++++++++[< +++> -]<.print E>> +++++++++++[< +++++++>-]<.print M>>++++++++[<+++++++++>-]<+.print I'
-        self.assertEqual(gityghasemi.execute(bfcode), 'GITY GHASEMI')
+        bf_code='>+++++++[<++++++++++> -]<+.print G>> ++++++++[< +++++++++>-]<+.print I>>+++++++++++++++++++++[<++++>-]< .print T>> +++++++++++[< ++++++++> -]<+.print Y>++++++++++++++++++++++++++++++++.print ''>>+++++++[< ++++++++++> -]<+.print G>> ++++++++[< +++++++++> -]<. print H>> +++++++++++++[< +++++> -]<.print A>> +++++++++++++++++++++[< ++++> -]< -.print S>> +++++++++++++++++++++++[< +++> -]<.print E>> +++++++++++[< +++++++>-]<.print M>>++++++++[<+++++++++>-]<+.print I'
+        self.assertEqual(gityghasemi.python_bf(bf_code), 'GITY GHASEMI')
         
     def test_errors(self):
         #test unbalance bracket
-        bfcode='++++[->++++++++<]------>>++>++<<<].'
-        self.assertRaises(Exception,gityghasemi.execute,bfcode)
-        
-        #test overflow pointer
-        bfcode='>'*700000 + '++++.'
-        self.assertRaises(Exception.gityghasemi.execute,bfcode)
+        bf_code='++++[->++++++++<]------>>++>++<<<].'
+        self.assertRaises(Exception,gityghasemi.python_bf,bf_code)
         
+                
         #test underflow pointer
-        bfcode='++++>++++<------.'
-        self.assertRaises(Exception.gityghasemi.execute,bfcode)
+        bf_code='++++>++++<------.'
+        self.assertRaises(Exception,gityghasemi.python_bf,bf_code)
         
         #test Memory overflow
-        bfcode='+'*256+'.'
-        self.assertRaises(Exception.gityghasemi.execute,bfcode)
+        bf_code='+'*256+'.'
+        self.assertRaises(Exception,gityghasemi.python_bf,bf_code)
         
         #test Memory underflow
-        bfcode='++++>++'+'<-----'
-        self.assertRaises(Exception.gityghasemi.execute,bfcode)
+        bf_code='++++>++'+'<-----.'
+        self.assertRaises(Exception,gityghasemi.python_bf,bf_code)
         
 
 if __name__ == '__main__':
-    unittest.main()
\ No newline at end of file
+    unittest.main()

```

Comment:
This cannot be right! You want to run unittest whenever your code is executed directly?

Unittest is not a part of your program. It is only to test the code. So, it doesn't make sense to run it within the code.